aws_xray_sdk/core/models/dummy_entities.py

```diff
@@ -12,9 +14,13 @@ class DummySegment(Segment):
     A dummy segment will not be sent to the X-Ray daemon. Manually create
     dummy segments is not recommended.
     """
-    def __init__(self, name='dummy'):
 
-        super(DummySegment, self).__init__(name=name, traceid=TraceId().to_id())
+    def __init__(self, name='dummy'):
+        noOpID = os.getenv('AWS_XRAY_NOOP_ID')
+        if not noOpID or noOpID.lower() == 'true':
```

Comment:
We should probably explicitly check for the `false` case, and generate a random ID if and only if the environment variable is set to false, since that's what our docs say. Looks like that's how we did it in Go too: https://github.com/aws/aws-xray-sdk-go/pull/293

Reply:
Here we can explicitly check for the `false` case but in the subsegment case we are checking for `true` so I have kept it here like this.